### DISCUSSION TITLE
Xephyr fails to start with `private-tmp` in firejail >= 0.9.60

### DISCUSSION THREAD

**Discussion by Roosn** [netblue30/firejail#4244] [Q&A] [answered]
Hello. After updating the version Firejail from 0.9.58.2 to 0.9.64.4 `Xephyr` does not start and gives errors:

`_XSERVTransSocketCreateListener: failed to bind listener`
`_XSERVTransSocketUNIXCreateListener: ...SocketCreateListener() failed`
`_XSERVTransMakeAllCOTSServerListeners: failed to create listener for unix`
`Error: failed to start xephyr`

When I turn off _(comment)_ `#private-tmp` everything works as before. Tell me, please, why on the old version it works with `private-tmp`, and in a new version is not work ?

If `private-tmp` replaced with `whitelist /tmp/.X11-unix`, everything works fine.

**Comment by rusty-snake** [comment]
`private-tmp` is just
<sub>([source code](https://github.com/netblue30/firejail/blob/190152f9484a0b3ab8244d1581dfe8a12d4ef7ae/src/firejail/fs.c#L1223))</sub>

```
whitelist /tmp/$XAUTHORITY
whitelist /tmp/.X11-unix
read-only /tmp/.X11-unix
whitelist /tmp/pulse-*
```

Does 

```
ignore read-only /tmp/.X11-unix
private-tmp
```
or
```
private-tmp
read-write /tmp/.X11-unix
```
work?

**Reply by Roosn** [reply]
Both examples do not work. But when I replacing the `private-tmp`on the code above and comment `read-only`:

```
#private-tmp
whitelist /tmp/$XAUTHORITY
whitelist /tmp/.X11-unix
#read-only /tmp/.X11-unix
whitelist /tmp/pulse-*
```
it works

**Reply by rusty-snake** [reply]
>  the code above 

Is just a pseudo representation and actually does not work the way it looks like.

> Both examples do not work.

😞

> But when I replacing the private-tmp

Where? (I should have asked this earlier)

**Reply by Roosn** [reply]
In profile `Xephyr.profile`

**Reply by Roosn** [reply]
For example, in `Xephyr.profile` only one srting `private-tmp` and nothing more. If I run `firejail --noprofile --x11=xephyr mousepad` Xephyr does not start. On the Firejail version `0.9.58.2` it works

**Reply by rusty-snake** [reply]
If you add a command to a profile (for further hardening) and the program fails with it, remove it.

**Reply by Roosn** [reply]
I try not to just remove, but choose a replacement (if possible). I hope in the future Xephyr with `private-tmp` will work again.

**Reply by rusty-snake** [reply]
You should mention it if you added a option that causes breakage.

**Comment by rusty-snake** [comment, marked as answer]
`private-tmp` mounts `/tmp/.X11-unix` `read-only` since feae44c4.
`private-tmp` is disable for Xephyr since a2801805.

According to the tests above, there is no way to make `/tmp/.X11-unix` rw with `private-tmp`. However, as you already discovered you can use ` whitelist /tmp/.X11-unix` to make Xephyr work with a private /tmp. As long as you don't have any xauth or pulse issues there is no further action needed.
